Neos.Fusion/Classes/FusionObjects/Helpers/LazySelfReferentialProps.php

```diff
@@ -0,0 +1,77 @@
+<?php
+declare(strict_types=1);
+namespace Neos\Fusion\FusionObjects\Helpers;
+
+/*
+ * This file is part of the Neos.Fusion package.
+ *
+ * (c) Contributors of the Neos Project - www.neos.io
+ *
+ * This package is Open Source Software. For the full copyright and license
+ * information, please view the LICENSE file which was distributed with this
+ * source code.
+ */
+
+use Neos\Flow\Annotations as Flow;
+use Neos\Fusion\Core\Runtime;
+
+/** @Flow\Proxy(false) */
+final class LazySelfReferentialProps implements \ArrayAccess, \Stringable
+{
+    /** @var array<string, LazyReference>  */
+    private array $lazyReferences = [];
+
+    public function __construct(
+        private string $parentPath,
+        private Runtime $runtime,
+        private array $effectiveContext,
+        private string $selfReferentialId
+    ) {
+        $this->effectiveContext[$this->selfReferentialId] = $this;
+    }
+
+    public function offsetGet($path): mixed
+    {
+        $fullyQualifiedPath = $this->parentPath . '/' . $path;
+
+        $this->lazyReferences[$path] ??= new LazyReference(
+            function () use ($fullyQualifiedPath) {
+                $this->runtime->pushContextArray($this->effectiveContext);
+                try {
+                    return $this->runtime->evaluate($fullyQualifiedPath);
```

Comment:
we can make it fail hard if we want to rely on the private prop to exist which sounds nice ...

Reply:
What do you mean?

Reply:
That you cannot do `private.nonExistant` but an exception will be thrown ;)